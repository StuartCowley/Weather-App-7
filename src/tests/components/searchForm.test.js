import React from "react";
import { render } from "@testing-library/react";
import SearchForm from "../../components /searchForm";
import handleCitySearch from "../../components /App";
import setSearchText from "../../components /App";

describe("SearchForms", () => {
  it("renders correctly", () => {
    const { asFragment } = render(
      <SearchForm
        onSubmit={handleCitySearch}
        searchText="Manchester"
        setSearchText={setSearchText}
      />
    );

    expect(asFragment()).toMatchSnapshot();
  });

  it("should render a button with the text 'search", () => {
    const { getByText } = render(
      <SearchForm
        onSubmit={handleCitySearch}
        searchText="Manchester"
        setSearchText={setSearchText}
      />
    );

    expect(getByText("Search")).toBeInTheDocument;
  });
});
