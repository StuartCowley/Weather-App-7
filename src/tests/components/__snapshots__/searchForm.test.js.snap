// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SearchForms renders correctly 1`] = `
<DocumentFragment>
  <div
    class="search-form"
  >
    <input
      type="text"
      value="Manchester"
    />
    <button
      type="submit"
    >
      Search
    </button>
  </div>
</DocumentFragment>
`;
